File: PeachTravel/Common/Common/TZTagsCollectionView/TZTagsCollectionView.m
    //
//  TZTagsCollectionView.m
//  PeachTravel
//
//  Created by liangpengshuai on 11/7/15.
//  Copyright © 2015 com.aizou.www. All rights reserved.
//

#import "TZTagsCollectionView.h"
#import "TZTagsCollectionViewCell.h"
#import "TaoziCollectionLayout.h"

@interface TZTagsCollectionView ()<UICollectionViewDataSource, UICollectionViewDelegate, TaoziLayoutDelegate>



@end

@implementation TZTagsCollectionView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        self.backgroundColor = [UIColor clearColor];
        _tagsList = @[@"货到付款", @"货到付款", @"货到付款"];
        self.delegate = self;
        self.dataSource = self;
        [self registerNib:[UINib nibWithNibName:@"TZTagsCollectionViewCell" bundle:nil] forCellWithReuseIdentifier:@"tzTagsCollectionViewCell"];
        TaoziCollectionLayout *layout = [[TaoziCollectionLayout alloc] init];
        self.collectionViewLayout = layout;
        self.scrollEnabled = NO;
        layout.delegate = self;
    }
    return self;
}

- (void)awakeFromNib
{
    [super awakeFromNib];
    _tagsList = @[@"货到付款", @"货到付款", @"货到付款"];
    self.delegate = self;
    self.dataSource = self;
    [self registerNib:[UINib nibWithNibName:@"TZTagsCollectionViewCell" bundle:nil] forCellWithReuseIdentifier:@"tzTagsCollectionViewCell"];
    TaoziCollectionLayout *layout =  (TaoziCollectionLayout *)self.collectionViewLayout;
    layout.spacePerItem = 10;
    layout.delegate = self;
    self.scrollEnabled = NO;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    return _tagsList.count;
}

- (NSInteger)numberOfItemsInSection:(NSInteger)section
{
    return 1;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    TZTagsCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"tzTagsCollectionViewCell" forIndexPath:indexPath];
    cell.tagLabel.text = _tagsList[indexPath.row];
    cell.tagLabel.layer.borderColor = APP_THEME_COLOR.CGColor;
    cell.tagLabel.textColor = APP_THEME_COLOR;

    return cell;
}


- (CGSize)tzCollectionView:(UICollectionView *)collectionView sizeForItemAtIndexPath:(NSIndexPath *)indexPath
{
    NSString *title = [_tagsList objectAtIndex:indexPath.row];
    CGSize size = [title sizeWithAttributes:@{NSFontAttributeName: [UIFont systemFontOfSize:9.0]}];
    return CGSizeMake(size.width+10, 16);
}

- (CGSize)tzCollectionview:(UICollectionView *)collectionView sizeForHeaderView:(NSIndexPath *)indexPath
{
    return CGSizeMake(kWindowWidth, 0);
}

- (NSInteger)tzNumberOfSectionsInTZCollectionView:(UICollectionView *)collectionView
{
    return 1;
}

- (NSInteger)tzCollectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    return _tagsList.count;
}

- (CGFloat)tzCollectionLayoutWidth
{
    return self.bounds.size.width;
}

@end

File: PeachTravel/Common/Common/CommonView/TZTagsCollectionView.swift
//
//  TZTagsCollectionView.swift
//  PeachTravel
//
//  Created by liangpengshuai on 10/17/15.
//  Copyright © 2015 com.aizou.www. All rights reserved.
//

import UIKit

class TZTagsCollectionView: UICollectionView, TaoziLayoutDelegate, UICollectionViewDelegate, UICollectionViewDataSource {
    
    var tagsList: Array<String>! {
        didSet {
            self.reloadData()
        }
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.backgroundColor = UIColor.whiteColor()
        tagsList = Array()
        if self.collectionViewLayout is TaoziCollectionLayout {
            let tzLayout: TaoziCollectionLayout = self.collectionViewLayout as! TaoziCollectionLayout
            tzLayout.delegate = self
        }
        self.delegate = self;
        self.dataSource = self;
        self.registerNib(UINib(nibName: "TZTagCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "tzTagCollectionCell")
    }
    
//    override init(frame: CGRect, collectionViewLayout layout: UICollectionViewLayout) {
//        tagsList = Array()
//        super.init(frame: frame, collectionViewLayout: layout)
//        if layout is TaoziCollectionLayout {
//            let tzLayout: TaoziCollectionLayout = layout as! TaoziCollectionLayout
//            tzLayout.delegate = self
//        }
//        self.delegate = self;
//        self.dataSource = self;
//        self.registerNib(UINib(nibName: "TZTagCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "tzTagCollectionCell")
//    }
    
    //MARK: UICollectionViewDataSource
    
    override func numberOfItemsInSection(section: Int) -> Int {
        return 1
    }
    
    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return tagsList.count
    }
    
    func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
        let cell: TZTagCollectionViewCell = collectionView.dequeueReusableCellWithReuseIdentifier("tzTagCollectionCell", forIndexPath: indexPath) as! TZTagCollectionViewCell
        cell.titleBackgroundColor = APP_THEME_COLOR
        cell.title = tagsList[indexPath.row]
        return cell
    }
    
    //MARK: TaoziLayoutDelegate
    func tzCollectionLayoutWidth() -> CGFloat {
        return self.bounds.size.width
    }
    
    func tzCollectionView(collectionView: UICollectionView!, numberOfItemsInSection section: Int) -> Int {
        return tagsList.count
    }
    
    func tzCollectionView(collectionView: UICollectionView!, sizeForItemAtIndexPath indexPath: NSIndexPath!) -> CGSize {
        return CGSizeMake(40, 15)
    }
    
    func tzNumberOfSectionsInTZCollectionView(collectionView: UICollectionView!) -> Int {
        return 1
    }

    func tzCollectionview(collectionView: UICollectionView!, sizeForHeaderView indexPath: NSIndexPath!) -> CGSize {
        return CGSizeZero
    }
}
